FreeRTOS_TCP_IP.c

```diff
@@ -3711,13 +3752,27 @@
         const TCPPacket_t * pxTCPPacket = ipCAST_CONST_PTR_TO_CONST_TYPE_PTR( TCPPacket_t, pxNetworkBuffer->pucEthernetBuffer );
         FreeRTOS_Socket_t * pxReturn = NULL;
         uint32_t ulInitialSequenceNumber;
+        struct freertos_sockaddr xAddress;
 
-        /* Assume that a new Initial Sequence Number will be required. Request
-         * it now in order to fail out if necessary. */
-        ulInitialSequenceNumber = ulApplicationGetNextSequenceNumber( *ipLOCAL_IP_ADDRESS_POINTER,
-                                                                      pxSocket->usLocalPort,
-                                                                      pxTCPPacket->xIPHeader.ulSourceIPAddress,
-                                                                      pxTCPPacket->xTCPHeader.usSourcePort );
+        /* Put the IP address in the address-structure. */
+        xAddress.sin_addr = pxTCPPacket->xIPHeader.ulDestinationIPAddress;
+
+        /* Silently discard a SYN packet which was sent to the broadcast address
+         * (x.x.x.255) or a multicast address. */
+        if( xIsUnicastAddress( &xAddress ) != pdFALSE )
```

Comment:
When we started this PR, we said that broadcast and multicast packets carrying a SYN flag should be silently discarded.
We already filter out other IP-addresses, only the device's IP address can be targeted.... and multi- and broadcast packets.

Now what if we don't call `xIsUnicastAddress()` but in stead a test like this?
~~~
if( pxTCPPacket->xIPHeader.ulDestinationIPAddress == *ipLOCAL_IP_ADDRESS_POINTER )
{
    ulInitialSequenceNumber = ulApplicationGetNextSequenceNumber( *ipLOCAL_IP_ADDRESS_POINTER );
}
else
{
    ulInitialSequenceNumber = 0U;
}
~~~
That makes the patch much smaller and more effective.

Reply:
I made this change but want to note that earlier change would have been more compatible with IPv6/multi.

Reply:
You're right about that, we could use the same code in both /single and in /multi